lib/src/signals/logic.dart

```diff
@@ -319,13 +319,59 @@ class Logic {
   void operator <=(Logic other) => gets(other);
 
   /// Logical bitwise NOT.
-  Logic operator ~() => NotGate(this).out;
+  Logic operator ~() {
+    if (this is Const) {
+      if (value.isValid) {
+        if (value == LogicValue.of(0, width: width)) {
+          return Const(LogicValue.filled(width, LogicValue.one));
+        } else if (value == LogicValue.filled(width, LogicValue.one)) {
+          return Const(LogicValue.of(0, width: width));
+        }
+      } else {
+        return Const(LogicValue.x, width: width);
+      }
+    }
+    return NotGate(this).out;
+  }
 
   /// Logical bitwise AND.
-  Logic operator &(Logic other) => And2Gate(this, other).out;
+  Logic operator &(Logic other) {
+    if (this is Const || other is Const) {
+      if (value.isValid && other.value.isValid) {
+        if (value == LogicValue.of(0, width: width) ||
+            other.value == LogicValue.of(0, width: width)) {
+          return Const(LogicValue.of(0, width: width));
+        } else if (value == LogicValue.filled(width, LogicValue.one)) {
+          return other;
+        } else if (other.value == LogicValue.filled(width, LogicValue.one)) {
+          return this;
+        }
+      } else {
+        return Const(LogicValue.x, width: width);
```

Comment:
This isn't quite right, consider 11x11 & 11111 == 11x11

If we just cover the cases of all 0's and all 1's that's probably enough and let the rest fall through to the and gate?